rend-spec-v3.txt

```diff
@@ -2334,31 +2324,51 @@ Appendix E. Managing authorized client data [CLIENT-AUTH-MGMT]
      human-readable authorized client names.
 
      Let's consider that one of the listed client names is "alice". In this
-     case, Tor checks the "client_authorized_pubkeys" file for any entries
-     with client_name being "alice". If an "alice" entry is found, we use the
-     relevant pubkeys to authenticate Alice.
+     case, Tor checks whether the "./client_authorized_privkeys/alice.keys"
+     file is present in the filesystem:
 
-     If no "alice" entry is found in the "client_authorized_pubkeys" file, Tor
-     is tasked with generating public/private keys for Alice. To do so, Tor
-     generates x25519 and ed25519 keypairs for Alice, then makes a
-     "client_authorized_privkeys/alice.privkey" file and writes the private
-     keys inside; it also adds an entry for alice to the
-     "client_authorized_pubkeys" file.
+     If the "alice.keys" file is found, we use the included keys to
+     authenticate Alice.
+
+     If no "alice.keys" file is found, Tor is tasked with generating Alice's
+     keypair. To do so, Tor generates x25519 and ed25519 keypairs for Alice,
+     then creates the "client_authorized_privkeys/alice.keys" file and writes
+     the private keys inside.
 
      In this last case, the hidden service operator has the responsibility to
      pass the .key file to Alice in a secure out-of-band way. After the file
      is passed to Alice, it can be shredded from the filesystem, as only the
      public keys are required for the hidden service to function.
 
-  E.1.2. Client side
+  E.1.2. Service-side bookkeeping
+
+     This section contains more details on how onion services should be keeping
+     track of the ".keys" file.
+
+     In particular, inside a ".keys" file, the onion service should be keeping
+     track of whether a key is ed25519 or x25519 and whether it's public or
+     private. An onion service SHOULD have either two private keys (one ed25519
+     and one x25519) or two public keys (one ed25519 and one x25519) in its
+     ".keys" file. The former scenario happens when the service generates the
+     client auth keys, whereas the latter scenario happens when the client
+     generates keys themselves and passes them to the service.
+
+     Here is a suggested scheme for "alice.keys" for the latter scenario:
+     """
+     alice
+     x25519 private c2247870536a192d142d056abefca68d6193158e7c1a59c1654c954eccaff894
+     ed25519 private 66c1a77104d86461b6f98f73acf3cd229c80624495d2d74d6fda1e940080a96b
+     """
+
+  E.1.3. Client side
 
      A client who wants to register client authorization data for a hidden service
      needs to add the following line to their torrc:
 
            HidServAuth onion-address x25519-private-key ed25519-private-key
```

Comment:
Do we want to put the keyword before the key like `HidServAuth onion-address x25519 [x25519-private-key] ed25519 [ed25519-private-key]` ?

Reply:
Might be a good idea I guess.